diff -ur a/lua-5.1.4/src/Makefile b/lua-5.1.4/src/Makefile
--- a/lua-5.1.4/src/Makefile	2008-01-20 05:37:58.000000000 +1000
+++ b/lua-5.1.4/src/Makefile	2009-12-15 20:05:21.000000000 +1000
@@ -52,10 +52,10 @@
 	$(RANLIB) $@
 
 $(LUA_T): $(LUA_O) $(LUA_A)
-	$(CC) -o $@ $(MYLDFLAGS) $(LUA_O) $(LUA_A) $(LIBS)
+	$(CCLD) -o $@ $(MYLDFLAGS) $(LUA_O) $(LUA_A) $(LIBS)
 
 $(LUAC_T): $(LUAC_O) $(LUA_A)
-	$(CC) -o $@ $(MYLDFLAGS) $(LUAC_O) $(LUA_A) $(LIBS)
+	$(CCLD) -o $@ $(MYLDFLAGS) $(LUAC_O) $(LUA_A) $(LIBS)
 
 clean:
 	$(RM) $(ALL_T) $(ALL_O)
