diff -ur a/newlib-1.17.0/newlib/libc/include/machine/setjmp.h b/newlib-1.17.0/newlib/libc/include/machine/setjmp.h
--- a/newlib-1.17.0/newlib/libc/include/machine/setjmp.h	2008-12-12 06:05:38.000000000 +1000
+++ b/newlib-1.17.0/newlib/libc/include/machine/setjmp.h	2009-12-22 19:37:52.000000000 +1000
@@ -60,7 +60,7 @@
 #if defined(__CYGWIN__) && !defined (_JBLEN)
 #define _JBLEN (13 * 4)
 #elif defined (__i386__)
-#if defined(__unix__) || defined(__rtems__)
+#if defined(__unix__) || defined(__rtems__) || defined(__APPLE__)
 # define _JBLEN        9
 #else
 #include "setjmp-dj.h"
@@ -245,6 +245,13 @@
 #define _JBTYPE unsigned short
 #endif /* __m32c__ */
 
+#ifdef __LLJVM__
+#ifdef _JBLEN
+#undef _JBLEN
+#endif
+#define _JBLEN 2
+#endif
+
 #ifdef _JBLEN
 #ifdef _JBTYPE
 typedef	_JBTYPE jmp_buf[_JBLEN];
diff -ur a/newlib-1.17.0/newlib/libc/include/sys/config.h b/newlib-1.17.0/newlib/libc/include/sys/config.h
--- a/newlib-1.17.0/newlib/libc/include/sys/config.h	2008-11-20 07:05:31.000000000 +1000
+++ b/newlib-1.17.0/newlib/libc/include/sys/config.h	2009-12-22 19:37:52.000000000 +1000
@@ -135,6 +135,11 @@
 #define __CUSTOM_FILE_IO__
 #endif
 
+#ifdef __LLJVM__
+#define _LONG_DOUBLE double
+#undef __LARGE64_FILES
+#endif
+
 /* This block should be kept in sync with GCC's limits.h.  The point
    of having these definitions here is to not include limits.h, which
    would pollute the user namespace, while still using types of the
diff -ur a/newlib-1.17.0/newlib/libc/include/sys/signal.h b/newlib-1.17.0/newlib/libc/include/sys/signal.h
--- a/newlib-1.17.0/newlib/libc/include/sys/signal.h	2006-04-19 06:06:09.000000000 +1000
+++ b/newlib-1.17.0/newlib/libc/include/sys/signal.h	2009-12-22 19:37:52.000000000 +1000
@@ -187,6 +187,10 @@
 
 #endif /* defined(__CYGWIN__) || defined(__rtems__) */
 
+#ifdef __LLJVM__
+int _EXFUN(kill, (int, int));
+#endif
+
 /* #endif __STRICT_ANSI__ */
 
 #if defined(___AM29K__)
diff -ur a/newlib-1.17.0/newlib/libc/include/sys/_types.h b/newlib-1.17.0/newlib/libc/include/sys/_types.h
--- a/newlib-1.17.0/newlib/libc/include/sys/_types.h	2007-09-19 06:05:17.000000000 +1000
+++ b/newlib-1.17.0/newlib/libc/include/sys/_types.h	2009-12-22 19:37:52.000000000 +1000
@@ -62,6 +62,11 @@
 #define __need_wint_t
 #include <stddef.h>
 
+#ifndef _WINT_T
+#define _WINT_T
+typedef unsigned int wint_t;
+#endif
+
 #ifndef __mbstate_t_defined
 /* Conversion state information.  */
 typedef struct
diff -ur a/newlib-1.17.0/newlib/libc/include/wchar.h b/newlib-1.17.0/newlib/libc/include/wchar.h
--- a/newlib-1.17.0/newlib/libc/include/wchar.h	2008-12-13 05:47:05.000000000 +1000
+++ b/newlib-1.17.0/newlib/libc/include/wchar.h	2009-12-22 19:37:52.000000000 +1000
@@ -10,6 +10,11 @@
 #define __need_wint_t
 #include <stddef.h>
 
+#ifndef _WINT_T
+#define _WINT_T
+typedef unsigned int wint_t;
+#endif
+
 /* For _mbstate_t definition. */
 #include <sys/_types.h>
 
diff -ur a/newlib-1.17.0/newlib/libc/include/wctype.h b/newlib-1.17.0/newlib/libc/include/wctype.h
--- a/newlib-1.17.0/newlib/libc/include/wctype.h	2002-11-07 10:44:31.000000000 +1000
+++ b/newlib-1.17.0/newlib/libc/include/wctype.h	2009-12-22 19:37:52.000000000 +1000
@@ -7,6 +7,11 @@
 #define __need_wint_t
 #include <stddef.h>
 
+#ifndef _WINT_T
+#define _WINT_T
+typedef unsigned int wint_t;
+#endif
+
 #ifndef WEOF
 # define WEOF ((wint_t)-1)
 #endif
diff -ur a/newlib-1.17.0/newlib/libc/reent/reent.c b/newlib-1.17.0/newlib/libc/reent/reent.c
--- a/newlib-1.17.0/newlib/libc/reent/reent.c	2006-10-11 18:04:50.000000000 +1000
+++ b/newlib-1.17.0/newlib/libc/reent/reent.c	2009-12-22 19:37:52.000000000 +1000
@@ -23,7 +23,7 @@
 
 /* We use the errno variable used by the system dependent layer.  */
 #undef errno
-int errno;
+extern int errno;
 
 #endif
 
diff -ur a/newlib-1.17.0/newlib/libc/reent/renamer.c b/newlib-1.17.0/newlib/libc/reent/renamer.c
--- a/newlib-1.17.0/newlib/libc/reent/renamer.c	2008-11-25 07:27:33.000000000 +1000
+++ b/newlib-1.17.0/newlib/libc/reent/renamer.c	2009-12-22 19:37:52.000000000 +1000
@@ -5,6 +5,10 @@
 #include <sys/stat.h>
 #include <_syslist.h>
 
+#ifdef HAVE_RENAME
+int _rename(const char *oldpath, const char *newpath);
+#endif
+
 /* Some targets provides their own versions of these functions.  Those
    targets should define REENTRANT_SYSCALLS_PROVIDED in TARGET_CFLAGS.  */
 
